### PR TITLE
Session timeout should save answers

### PR DESCRIPTION
**Pull request by benjefferies**
### What is the context of this PR?
Session timeout should save answers when user clicks save and sign out. 
If a user times out before clicking save and sign out all answers but the latest page they had timed out on should be saved

### How to review 
Verify timeout.spec.js tests the scenario correctly

### CODE REVIEW

File: app/views/questionnaire.py [hunk 1]
```diff
@@ -292,7 +292,6 @@ def get_timeout_continue(eq_id, form_type, collection_id):  # pylint: disable=un
 @questionnaire_blueprint.route('session-expired', methods=["GET"])
 @login_required
 def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
-    _delete_user_data()
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py:207]: you should replace this with logout_user() to kill the flask session

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/views/questionnaire.py (hunk 1 above)
Comment (benjefferies) [reply app/views/questionnaire.py:207]: Good spot, I'll also add an IT for it

File: app/views/questionnaire.py [hunk 2]
```diff
@@ -292,6 +292,7 @@ def get_timeout_continue(eq_id, form_type, collection_id):  # pylint: disable=un
 @questionnaire_blueprint.route('session-expired', methods=["GET"])
 @login_required
 def get_session_expired(eq_id, form_type, collection_id):  # pylint: disable=unused-argument
+    session_storage.clear()
```
Comment (jonnyshaw89) [inline app/views/questionnaire.py]: i still think we need to call logout_user if only just to stop a login attempt when they reload?

**Review by LJBabbage — APPROVED** [review]

**Comment by LJBabbage** [comment]
Stick a "fixes" in the PR text so it closes the issue when its merged

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by collisdigital** [comment]
Would be good if this was in for the demo tomorrow!

**Comment by benjefferies** [comment]
Tell travis that @collisdigital. Fix one intermittent timeout test another one fails.

**Comment by benjefferies** [comment]
I lied. It was broken from the rebase. test_timeout.json now needs a summary block.

**Comment by collisdigital** [comment]
👍

**Comment by collisdigital** [comment]
Fixes #991

**Pull request merged**